middleware/http/span.go

```diff
@@ -0,0 +1,30 @@
+package http
+
+import (
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// NewSpanMiddleware returns a middleware function that creates a span for its requests.
+// It takes the `spanName` and `attributes` as an argument. The attributes are essentially key value tuples used to set
+// the span's attributes during creation.
+// This middleware must be used in conjunction after the `NewTracedMiddleware`
+func NewSpanMiddleware(spanName string, attributes ...attribute.KeyValue) func(next http.Handler) http.Handler {
```

Comment:
@shan-96 I think it should be request based middleware, not sure that how connection based span will work. 
- from one side connection live time can be very long, and not sure it make sense for keeping all requests in single span.
- but based on implementation span will be ended as soon as we upgrade connection.